scipy/optimize/_basinhopping.py

```diff
@@ -333,18 +347,17 @@ def accept_reject(self, energy_new, energy_old):
             #
             # Ignore this warning so when the algorithm is on a flat plane, it always
             # accepts the step, to try to move off the plane.
-            prod = -(energy_new - energy_old) * self.beta
+            prod = -(res_new.fun - res_old.fun) * self.beta
             w = math.exp(min(0, prod))
 
         rand = self.random_gen.uniform()
-        return w >= rand
+        return w >= rand and (res_new.success or not res_old.success)
```

Comment:
The default `Metropolis` should not accept a step with unsuccessful local search unless the incumbent also was the result of unsuccessful local search. (In the latter case, we don't _have_ to accept the step, but why not? It might help to explore the space.)